README.md

```diff
@@ -2,31 +2,739 @@
+```text
+count
+
+3
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Count the number of dogs:
+
+```sql
+SELECT COUNT(*)
+FROM pet
+WHERE species = 'dog';
+```
+
+</td>
+<td>
+
+```text
+count
+
+4
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Sum the ages of all pets:
+
+```sql
+SELECT SUM(age)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+sum
+
+35
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Sum the ages of Tabby cats:
+
+```sql
+SELECT SUM(age)
+FROM pet
+WHERE breed = 'Tabby' and species = 'cat' ;
+```
+
+</td>
+<td>
+
+```text
+sum
+
+11
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Minimum, maximum and average age
+among all pets:
+
+```sql
+SELECT MIN(age), MAX(age), AVG(age)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+min max avg
+
+1   12  5    
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Minimum, maximum and average age
+among all cats:
+
+```sql
+SELECT MIN(age), MAX(age), AVG(age)
+FROM pet
+WHERE species = 'cat';
+```
+
+</td>
+<td>
+
+```text
+min max avg
+
+1   10  5.333333333
+```
+
+</td>
+</tr>
+
+
+<tr>
+<td>
+Truncate to display 2 digits after decimal point:
+
+```sql
+SELECT MIN(age), MAX(age), TRUNC(AVG(age),2) 
+FROM pet
+WHERE species = 'cat';
+```
+
+</td>
+<td>
+
+```text
+min max avg
+
+1   10  5.33
+```
+
+</td>
+</tr>
+
+</table>
+
+
+
+## GROUP BY
+
+The `GROUP BY` clause collects rows with the same value for a column into a group.
+
+
+```text
+id  name        species	breed	            age
+
+1   Moe         cat     Tabby	            10
+2   Hana        cat     Tabby	            1
+3   Lil' Bub    cat     American Shorthair  5
+4   Snuggles    dog     Bichon	            5
+5   Honey       dog     Cavachon            NULL
+6   Herbie      fish    NULL                4
+7   Fifi        dog     Poodle              1
+8   Lulu        dog     Yorkiepoo           12
+```
+
+
+For example, grouping the `pet` table by `species` creates three groups of rows:
+
+- cat group: rows with id 1, 2, and 3
+- dog group: rows with id 4, 5, 7, and 8
+- fish group: row with id 6
+
+We often use the `GROUP BY` clause with an aggregate function
+to calculate a measure for each group.  The `SELECT` clause
+should list the grouped column, along with one or more aggregate
+functions.  
+
+NOTE: Your row order may differ from the output displayed below.
+
+<table>
+<tr>
+<th>
+Query
+</th>
+<th>
+Result
+</th>
+</tr>
+
+<tr>
+<td>
+Display a count of rows per species:
+
+```sql
+SELECT species, COUNT(*)
+FROM pet
+GROUP BY species
+```
+
+</td>
+<td>
+
+```text
+species count
+
+dog     4
+cat     3
+fish    1
+```
+
+</td>
+</tr>
+
+
+<tr>
+<td>
+Sort the species count
+in ascending order:
+
+```sql
+SELECT species, COUNT(*)
+FROM pet
+GROUP BY species
+ORDER BY count
+```
+
+</td>
+<td>
+
+```text
+species count
+
+fish    1
+cat     3
+dog     4
+```
+
+</td>
+</tr>
+
+
+
+<tr>
+<td>
+Count rows per age,
+omitting null values:
+
+```sql
+SELECT age, COUNT(*)
+FROM pet
+WHERE age IS NOT NULL
+GROUP BY age
+```
+
+</td>
+<td>
+
+```text
+age count
+
+5   1
+4   1
+10  1
+2   1
+12  1
+1   2
```

Comment:
Double check this query. I'm getting:

age    count
5        2
4        1
10      1
12      1
1        2

Reply:
yes, updated.